### ISSUE TITLE
Operator doesn't apply pod's ObjectMeta defined on the CustomPodAutoscaler CR.

### ISSUE THREAD

**Issue by StefanCenusa** [jthomperoo/custom-pod-autoscaler-operator#87]
**Describe the bug**
Operator doesn't apply pod's ObjectMeta defined on the CustomPodAutoscaler CR.

**To Reproduce**
1. Deploy this CPA
```
apiVersion: custompodautoscaler.com/v1
kind: CustomPodAutoscaler
metadata:
  name: cpa-test
spec:
  template:
    metadata:
      name: cpa-test-pod
      labels:
        test: "true"
...
````
2. kubectl get po
3. See pod with name `cpa-test` insead of `cpa-test-pod`

**Expected behavior**
The pod should have the name specified in the PodTemplateSpec.

**Kubernetes Details (`kubectl version`):**
1.21

**Custom Pod Autoscaler Operator Details:**
v1.2.0

**Additional context**
The issue I think is that the CPA custom resource [openapi schema](https://github.com/jthomperoo/custom-pod-autoscaler-operator/blob/master/helm/templates/crd/custompodautoscaler.com_custompodautoscalers.yaml#L79-L85) doesn't define the values for PodSpecTemplate's metadata, thus the CPA CR is stored incompletely by the kube api-server.

```
❯ k get cpa -o yaml cpa-test
apiVersion: custompodautoscaler.com/v1
kind: CustomPodAutoscaler
metadata:
  name: cpa-test
spec:
  config:
   ...
  roleRequiresMetricsServer: true
  scaleTargetRef:
    apiVersion: apps/v1
    kind: Deployment
    name: hello-kubernetes
  template:
    metadata: {}
    spec:
      containers:
      - image: ...
        imagePullPolicy: IfNotPresent
        name: cpa-test
      terminationGracePeriodSeconds: 5
```` 
notice the `metadata: {}` although I've defined it on the applied CPA resource

**Comment by StefanCenusa** [comment]
I've tested with the following change:

```
              template:
                description: The image of the Custom Pod Autoscaler
                properties:
                  metadata:
                    description: 'Standard object''s metadata. More info: https://git.k8s.io/community/contributors/devel/sig-architecture/api-conventions.md#metadata'
                    type: object
                    x-kubernetes-preserve-unknown-fields: true
```
And it works. Metadata defined on CPA is applied to the created Pod

**Comment by StefanCenusa** [comment]
@jthomperoo When will this change be released?

**Comment by jthomperoo** [comment]
Now available in [`1.3.0`](https://github.com/jthomperoo/custom-pod-autoscaler-operator/releases/tag/v1.3.0) 🎉
